src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Specific.cs

```diff
@@ -115,54 +115,57 @@ public void ConnectWithConflictingDirections_Throws_UnauthorizedAccessException(
-                    Assert.Equal(expectedRead, await server.ReadAsync(received6, msg1.Length - expectedRead, expectedRead));
-                    Assert.True(server.IsMessageComplete);
-                    Assert.Equal(msg1, received6);
+                        server.WaitForConnection();
+
+                        int len1 = server.Read(received1, 0, msg1.Length);
+                        Assert.True(server.IsMessageComplete);
+                        Assert.Equal(msg1.Length, len1);
+                        Assert.Equal(msg1, received1);
+
+                        int len2 = server.Read(received2, 0, msg2.Length);
+                        Assert.True(server.IsMessageComplete);
+                        Assert.Equal(msg2.Length, len2);
+                        Assert.Equal(msg2, received2);
+
+                        int expectedRead = msg1.Length - 1;
+                        int len3 = server.Read(received3, 0, expectedRead);  // read one less than message
+                        Assert.False(server.IsMessageComplete);
+                        Assert.Equal(expectedRead, len3);
+                        for (int i = 0; i < expectedRead; ++i)
+                        {
+                            Assert.Equal(msg1[i], received3[i]);
+                        }
+
+                        expectedRead = msg1.Length - expectedRead;
+                        Assert.Equal(expectedRead, server.Read(received3, len3, expectedRead));
+                        Assert.True(server.IsMessageComplete);
+                        Assert.Equal(msg1, received3);
+
+                        Assert.Equal(msg1.Length, await server.ReadAsync(received4, 0, msg1.Length));
+                        Assert.True(server.IsMessageComplete);
+                        Assert.Equal(msg1, received4);
+
+                        Assert.Equal(msg2.Length, await server.ReadAsync(received5, 0, msg2.Length));
+                        Assert.True(server.IsMessageComplete);
+                        Assert.Equal(msg2, received5);
+
+                        expectedRead = msg1.Length - 1;
+                        Assert.Equal(expectedRead, await server.ReadAsync(received6, 0, expectedRead));  // read one less than message
+                        Assert.False(server.IsMessageComplete);
+                        for (int i = 0; i < expectedRead; ++i)
+                        {
+                            Assert.Equal(msg1[i], received6[i]);
+                        }
+
+                        expectedRead = msg1.Length - expectedRead;
+                        Assert.Equal(expectedRead, await server.ReadAsync(received6, msg1.Length - expectedRead, expectedRead));
+                        Assert.True(server.IsMessageComplete);
+                        Assert.Equal(msg1, received6);
+                    });
 
-                    await clientTask;
+                    Assert.True(Task.WaitAll(new[] { clientTask, serverTask }, TimeSpan.FromSeconds(15)));
```

Comment:
note you could leave the method async and instead ahve `await Task.WhenAll( ...`. So xunit can continue. But it means no timeout. Not sure which is best, probably this is fine

Reply:
We need the timeout the steps on the test can block